### DISCUSSION TITLE
Panel routes registrations

### DISCUSSION THREAD

**Discussion by adrien-delhom** [filamentphp/filament#12446] [Ideas]
Hi 👋

Currently, we don't really have control over how our panels routing are registered.
I wonder if it would be interesting to have an approach a bit like that in /routes/web.php

```php
Route::name('filament.')
    ->group(function () {
        foreach (Filament::getPanels() as $panel) {
            $panel->registerRoutes(); # define in a trait Panel\Concerns\HasRouteRegistration
        }
    });
```

This would allow everyone to override the way their panel routes are registered.

What do you think, guys? Is it worth doing a PR ? Do any of you have this need ?

**Comment by fpolli** [comment]
Where does Filament define routes? I'm used to working in default Laravel where your routes are defined in ... route files and endpoints go to controllers. But obviously Filament does not use route files and components are their own endpoints. I am getting a "Route [filament.admin.auth.register] not defined" error, and route:list shows that indeed it is not defined, but I have no idea where to look or where to have it defined.

**Comment by begueradj** [comment]
Any updates about this ?
